src/Core/Domain/Product/AttributeGroup/Query/GetProductAttributeGroups.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\AttributeGroup\Query;
+
+use PrestaShop\PrestaShop\Core\Domain\Language\Exception\LanguageException;
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+
+/**
+ * Query which provides attributes of a Product by group
+ */
+class GetProductAttributeGroups
+{
+    /**
+     * @var ProductId
+     */
+    private $productId;
+
+    /**
+     * @var LanguageId
+     */
+    private $languageId;
```

Comment:
I feel like we shouldn't limit the value to a Language (from a command point of view) but return localized values
Even if in the interface we will likely use only the context Language (no need to switch these data when we switch language in the page since these values are not editable)

I had the same question for combination list (which has the same principle), but for combinations maybe it would be interesting for the merchant to see the Combination name change Although it might be harder for him to identify the combination he is searching for (but he can always switch back to his favorite language)